cmake/Linker.cmake

```diff
@@ -0,0 +1,33 @@
+option(ENABLE_USER_LINKER "Enable a specific linker if available" OFF)
+
+include(CheckCXXCompilerFlag)
+
+set(USER_LINKER_OPTION
+    "lld"
+    CACHE STRING "Linker to be used")
+set(USER_LINKER_OPTION_VALUES "lld" "gold" "bfd")
+set_property(CACHE USER_LINKER_OPTION PROPERTY STRINGS ${USER_LINKER_OPTION_VALUES})
+list(
+  FIND
+  USER_LINKER_OPTION_VALUES
+  ${USER_LINKER_OPTION}
+  USER_LINKER_OPTION_INDEX)
+
+if(${USER_LINKER_OPTION_INDEX} EQUAL -1)
+  message(
+    STATUS
+      "Using custom linker: '${USER_LINKER_OPTION}', explicitly supported entries are ${USER_LINKER_OPTION_VALUES}")
+endif()
+
+function(configure_linker)
+    if(NOT ENABLE_USER_LINKER)
+        return()
+    endif()
+
+    set(LINKER_FLAG "-fuse-ld=${USER_LINKER_OPTION}")
+
+    check_cxx_compiler_flag(${LINKER_FLAG} CXX_SUPPORTS_USER_LINKER)
+    if(CXX_SUPPORTS_USER_LINKER)
+        add_compile_options(${LINKER_FLAG})
```

Comment:
Quick question about this: 

Would there be any reason to use `target_compile_options` here instead, so that different targets could use a different linker? 

The `add_compile_options` function is usually a red flag for me, because it tends to make CMake code less flexible. On the other hand, I can't think of any reason you would want to use one linker for one target and a different one for another.

Reply:
my fault, moving to `target_compile_options`